addons/loyalty/models/loyalty_program.py

```diff
@@ -0,0 +1,315 @@
+            raise UserError(_('You can not delete a program in an active state'))
+
+    def toggle_active(self):
+        super().toggle_active()
+        # Propagate active state to children
+        for program in self:
+            program.rule_ids.active = program.active
+            program.reward_ids.active = program.active
+            program.communication_plan_ids = program.active
+            program.reward_ids.discount_line_product_id.active = program.active
+
+    @api.model
+    def get_program_templates(self):
+        '''
+        Returns the templates to be used for promotional programs.
+        '''
+        return {
+            'promo_code': {
+                'title': _('Promo Code'),
+                'description': _('Get a code to receive 10% discount on specific products'),
+            },
+            'gift_card': {
+                'title': _('Gift Card'),
+                'description': _('Sell Gift Cards, that can be used to purchase products'),
+            },
+            'loyalty': {
+                'title': _('Loyalty Cards'),
+                'description': _('Win points with each purchases, and use points to get gifts'),
+            },
+            'fidelity': {
+                'title': _('Fidelity Cards'),
+                'description': _('Buy 10 products, and get 10$ discount on the 11th one'),
+            },
+            'promotion': {
+                'title': _('Promotional Program'),
+                'description': _('Automatic promotion: free shipping on orders higher than $50'),
+            },
+            'coupons': {
+                'title': _('Coupons'),
+                'description': _('Send unique coupons that give access to rewards'),
+            },
+            'buy_two_get_one': {
+                'title': _('2+1 Free'),
+                'description': _('Buy 2 products and get a third one for free'),
+            },
+            'ewallet': {
+                'title': _('eWallet'),
+                'description': _('Fill in your eWallet, and use it to pay future orders'),
+            },
+        }
+
+    @api.model
+    def create_from_template(self, template_id):
+        '''
+        Creates the program from the template id defined in `get_program_templates`.
+
+        Returns an action leading to that new record.
+        '''
+        template_values = self._get_template_values()
+        if template_id not in template_values:
+            return False
+        program = self.create(template_values[template_id])
+        action = self.env['ir.actions.act_window']._for_xml_id('loyalty.loyalty_program_action')
+        action['view_type'] = 'form'
+        action['res_id'] = program.id
+        return action
+
+    @api.model
+    def _get_template_values(self):
+        '''
+        Returns the values to create a program using the template keys defined above.
+        '''
+        return {
+            'promo_code': {
+                'name': _('Promo Code'),
```

Comment:
Using lazy translations will probably improve the performances over here, same for ``get_program_templates``
```suggestion
                'name': _lt('Promo Code'),
```